### ISSUE TITLE
Atom hangs on opening multiple same name files with similar paths

### ISSUE THREAD

**Issue by alutman** [atom/atom#10043]
When trying to open multiple files with the same name but under slightly different paths, atom crashes.

**Reproduction Steps:**

**1**. Create the following folder and file structure outside of Atom. Files do not need to have any content.
```
root/
      folder1/ 
             sub1/
                  file.txt
             sub2/
                  file.txt
      folder2/ 
             sub1/
                  file.txt
             sub2/
                  file.txt
````
**2**. Open the folder `root` in Atom using File -> Open Folder
**3**. Open any three files by clicking on them in the Tree View

**Expected behavior:**
Atom opens up all three files in their own tabs

**Observed behavior:**
Atom editor stops responding on trying to open the third file, a few seconds later a dialog pops up asking to wait or close. Wait never resolves so closing is the only cause of action. The developer console shows no exception.

**Atom version:** 1.3.1
**OS and version:** Windows 7

**Installed packages:**
None

**Additional information:**
This bug occurs on a fresh install of atom with zero config changes or package installs.
I believe this has something to do with how the tabs are named with files that have the same name. 
When opening two files you get one of the following tab names

* `folder1\...\file.txt` | `folder2\...\file.txt`
* `sub1\file.txt` | `sub2\file.txt`

However when you open a third file, Atom doesn't know how to name the 3rd tab because if it tries to follow one of the formats above, it will have two files that have identical tab names. e.g.

`folder1\...\file.txt` (Meaning folder1\sub1\file.txt) | `folder2\...\file.txt` (Meaning folder2\sub1\file.txt) | `folder1\...\file.txt` (Meaning folder1\sub**2**\file.txt)

* Problem can be reproduced in safe mode: **Yes**
* Problem started happening recently, didn't happen in an older version of Atom: **Yes**
    * Can't reproduce this problem in 1.2.3 Windows 7 or 1.2.0 Linux Debian. It's worth noting that 1.2 formats the tab names differently (`file.txt - folder1\sub1`)
* Problem can be reliably reproduced, doesn't happen randomly: **Yes**
* Problem happens with all files and projects, not only some files or projects: **Yes**

**Comment by korgon** [comment]
I too am seeing this issue occur when I open multiple files (index.js) in this case.

Atom v1.3.1
Gentoo Linux

**Comment by lee-dohm** [comment]
I can't reproduce this using Atom v1.5.0-dev-f6da4a9 or Atom v1.4.0-beta0 on Mac OS X 10.11.2. The tab labeling scheme on both versions is the one described as belonging to v1.2.

**Comment by alutman** [comment]
I also cannot reproduce using the Windows 7 1.4.0-beta0

If the bug has been fixed in the upcoming beta, should we close the issue?

**Comment by lee-dohm** [comment]
Yes, I suspect that since the file naming scheme changed and then changed back that someone else ran across the bug and reverted whatever the change was. I'll see if I can track down the change and if maybe we can get a patch release out to Stable with it.

Thanks for reporting the issue!

**Comment by mtree** [comment]
Confirming
```
« mtree : ~ »» atom --version
1.3.1
```

**Comment by Falieson** [comment]
$  sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.2
BuildVersion:	15C50
$ atom --version
1.3.1

**Comment by 50Wliu** [comment]
This has already been fixed in the 1.4.0 betas.

**Comment by dave-handy** [comment]
Phew. Glad I found this ticket. The suggested ansible project layout has like every file named as 'main.yml' so atom has been awful.
